src/main/java/net/rptools/common/expression/ExpressionParser.java

```diff
@@ -207,8 +207,11 @@ public Result evaluate(String expression) throws ParserException {
     Result ret = new Result(expression);
     RunData oldData = RunData.hasCurrent() ? RunData.getCurrent() : null;
     try {
-      RunData newRunData = new RunData(ret);
-      RunData.setCurrent(newRunData);
+      RunData newRunData = oldData;
+      if (oldData == null) {
+        newRunData = new RunData(ret);
+        RunData.setCurrent(newRunData);
+      }
```

Comment:
This is my main concern with this change - it wasn't clear to me what the original code was hoping to accomplish by discarding the previously configured `current` RunData.  All tests continue to succeed in the wake of this change, but if someone has some insight into the original goals here it would help confirm that this modification is valid.